### ISSUE TITLE
Missing `overrides` in `XWindowsScreen` on FreeBSD

### ISSUE THREAD

**Issue by yurivict** [symless/synergy#7457]
### Discussion

- [X] Yes, and I still want to report a bug

### What type of Synergy user are you?

Community user

### Product name

Synergy 1

### Synergy version number

1.15.1

### Operating systems (OS)

- [ ] Windows
- [ ] macOS
- [ ] Linux or Unix
- [X] Other

### OS versions or Linux distros

FreeBSD 14.1

### Synergy configuration

FreeBSD 14.1

### What steps will reproduce the problem?

Regular build.

### Log output

```shell
In file included from /usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/lib/platform/XWindowsScreen.cpp:19:
/usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/./lib/platform/XWindowsScreen.h:55:17: error: 'getEventTarget' overrides a member function but is not marked 'override' [-Werror,-Winconsistent-missing-override]
   55 |   virtual void *getEventTarget() const;
      |                 ^
/usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/./lib/synergy/PlatformScreen.h:40:17: note: overridden virtual function is here
   40 |   virtual void *getEventTarget() const = 0;
      |                 ^
In file included from /usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/lib/platform/XWindowsScreen.cpp:19:
/usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/./lib/platform/XWindowsScreen.h:56:16: error: 'getClipboard' overrides a member function but is not marked 'override' [-Werror,-Winconsistent-missing-override]
   56 |   virtual bool getClipboard(ClipboardID id, IClipboard *) const;
      |                ^
/usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/./lib/synergy/PlatformScreen.h:41:16: note: overridden virtual function is here
   41 |   virtual bool getClipboard(ClipboardID id, IClipboard *) const = 0;
      |                ^
In file included from /usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/lib/platform/XWindowsScreen.cpp:19:
/usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/./lib/platform/XWindowsScreen.h:58:3: error: 'getShape' overrides a member function but is not marked 'override' [-Werror,-Winconsistent-missing-override]
   58 |   getShape(SInt32 &x, SInt32 &y, SInt32 &width, SInt32 &height) const;
      |   ^
/usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/./lib/synergy/PlatformScreen.h:43:3: note: overridden virtual function is here
   43 |   getShape(SInt32 &x, SInt32 &y, SInt32 &width, SInt32 &height) const = 0;
      |   ^
In file included from /usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/lib/platform/XWindowsScreen.cpp:19:
/usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/./lib/platform/XWindowsScreen.h:59:16: error: 'getCursorPos' overrides a member function but is not marked 'override' [-Werror,-Winconsistent-missing-override]
   59 |   virtual void getCursorPos(SInt32 &x, SInt32 &y) const;
      |                ^
/usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/./lib/synergy/PlatformScreen.h:44:16: note: overridden virtual function is here
   44 |   virtual void getCursorPos(SInt32 &x, SInt32 &y) const = 0;
      |                ^
In file included from /usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/lib/platform/XWindowsScreen.cpp:19:
/usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/./lib/platform/XWindowsScreen.h:62:16: error: 'reconfigure' overrides a member function but is not marked 'override' [-Werror,-Winconsistent-missing-override]
   62 |   virtual void reconfigure(UInt32 activeSides);
      |                ^
/usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/./lib/synergy/PlatformScreen.h:47:16: note: overridden virtual function is here
   47 |   virtual void reconfigure(UInt32 activeSides) = 0;
      |                ^
In file included from /usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/lib/platform/XWindowsScreen.cpp:19:
/usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/./lib/platform/XWindowsScreen.h:63:16: error: 'warpCursor' overrides a member function but is not marked 'override' [-Werror,-Winconsistent-missing-override]
   63 |   virtual void warpCursor(SInt32 x, SInt32 y);
      |                ^
/usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/./lib/synergy/PlatformScreen.h:48:16: note: overridden virtual function is here
   48 |   virtual void warpCursor(SInt32 x, SInt32 y) = 0;
      |                ^
In file included from /usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/lib/platform/XWindowsScreen.cpp:19:
/usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/./lib/platform/XWindowsScreen.h:64:18: error: 'registerHotKey' overrides a member function but is not marked 'override' [-Werror,-Winconsistent-missing-override]
   64 |   virtual UInt32 registerHotKey(KeyID key, KeyModifierMask mask);
      |                  ^
/usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/./lib/synergy/PlatformScreen.h:49:18: note: overridden virtual function is here
   49 |   virtual UInt32 registerHotKey(KeyID key, KeyModifierMask mask) = 0;
      |                  ^
In file included from /usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/lib/platform/XWindowsScreen.cpp:19:
/usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/./lib/platform/XWindowsScreen.h:65:16: error: 'unregisterHotKey' overrides a member function but is not marked 'override' [-Werror,-Winconsistent-missing-override]
   65 |   virtual void unregisterHotKey(UInt32 id);
      |                ^
/usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/./lib/synergy/PlatformScreen.h:50:16: note: overridden virtual function is here
   50 |   virtual void unregisterHotKey(UInt32 id) = 0;
      |                ^
In file included from /usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/lib/platform/XWindowsScreen.cpp:19:
/usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/./lib/platform/XWindowsScreen.h:66:16: error: 'fakeInputBegin' overrides a member function but is not marked 'override' [-Werror,-Winconsistent-missing-override]
   66 |   virtual void fakeInputBegin();
      |                ^
/usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/./lib/synergy/PlatformScreen.h:51:16: note: overridden virtual function is here
   51 |   virtual void fakeInputBegin() = 0;
      |                ^
In file included from /usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/lib/platform/XWindowsScreen.cpp:19:
/usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/./lib/platform/XWindowsScreen.h:67:16: error: 'fakeInputEnd' overrides a member function but is not marked 'override' [-Werror,-Winconsistent-missing-override]
   67 |   virtual void fakeInputEnd();
      |                ^
/usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/./lib/synergy/PlatformScreen.h:52:16: note: overridden virtual function is here
   52 |   virtual void fakeInputEnd() = 0;
      |                ^
In file included from /usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/lib/platform/XWindowsScreen.cpp:19:
/usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/./lib/platform/XWindowsScreen.h:68:18: error: 'getJumpZoneSize' overrides a member function but is not marked 'override' [-Werror,-Winconsistent-missing-override]
   68 |   virtual SInt32 getJumpZoneSize() const;
      |                  ^
/usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/./lib/synergy/PlatformScreen.h:53:18: note: overridden virtual function is here
   53 |   virtual SInt32 getJumpZoneSize() const = 0;
      |                  ^
In file included from /usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/lib/platform/XWindowsScreen.cpp:19:
/usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/./lib/platform/XWindowsScreen.h:69:16: error: 'isAnyMouseButtonDown' overrides a member function but is not marked 'override' [-Werror,-Winconsistent-missing-override]
   69 |   virtual bool isAnyMouseButtonDown(UInt32 &buttonID) const;
      |                ^
/usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/./lib/synergy/PlatformScreen.h:54:16: note: overridden virtual function is here
   54 |   virtual bool isAnyMouseButtonDown(UInt32 &buttonID) const = 0;
      |                ^
In file included from /usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/lib/platform/XWindowsScreen.cpp:19:
/usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/./lib/platform/XWindowsScreen.h:70:16: error: 'getCursorCenter' overrides a member function but is not marked 'override' [-Werror,-Winconsistent-missing-override]
   70 |   virtual void getCursorCenter(SInt32 &x, SInt32 &y) const;
      |                ^
/usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/./lib/synergy/PlatformScreen.h:55:16: note: overridden virtual function is here
   55 |   virtual void getCursorCenter(SInt32 &x, SInt32 &y) const = 0;
      |                ^
In file included from /usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/lib/platform/XWindowsScreen.cpp:19:
/usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/./lib/platform/XWindowsScreen.h:73:16: error: 'fakeMouseButton' overrides a member function but is not marked 'override' [-Werror,-Winconsistent-missing-override]
   73 |   virtual void fakeMouseButton(ButtonID id, bool press);
      |                ^
/usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/./lib/synergy/PlatformScreen.h:58:16: note: overridden virtual function is here
   58 |   virtual void fakeMouseButton(ButtonID id, bool press) = 0;
      |                ^
In file included from /usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/lib/platform/XWindowsScreen.cpp:19:
/usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/./lib/platform/XWindowsScreen.h:74:16: error: 'fakeMouseMove' overrides a member function but is not marked 'override' [-Werror,-Winconsistent-missing-override]
   74 |   virtual void fakeMouseMove(SInt32 x, SInt32 y);
      |                ^
/usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/./lib/synergy/PlatformScreen.h:59:16: note: overridden virtual function is here
   59 |   virtual void fakeMouseMove(SInt32 x, SInt32 y) = 0;
      |                ^
In file included from /usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/lib/platform/XWindowsScreen.cpp:19:
/usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/./lib/platform/XWindowsScreen.h:75:16: error: 'fakeMouseRelativeMove' overrides a member function but is not marked 'override' [-Werror,-Winconsistent-missing-override]
   75 |   virtual void fakeMouseRelativeMove(SInt32 dx, SInt32 dy) const;
      |                ^
/usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/./lib/synergy/PlatformScreen.h:60:16: note: overridden virtual function is here
   60 |   virtual void fakeMouseRelativeMove(SInt32 dx, SInt32 dy) const = 0;
      |                ^
In file included from /usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/lib/platform/XWindowsScreen.cpp:19:
/usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/./lib/platform/XWindowsScreen.h:76:16: error: 'fakeMouseWheel' overrides a member function but is not marked 'override' [-Werror,-Winconsistent-missing-override]
   76 |   virtual void fakeMouseWheel(SInt32 xDelta, SInt32 yDelta) const;
      |                ^
/usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/./lib/synergy/PlatformScreen.h:61:16: note: overridden virtual function is here
   61 |   virtual void fakeMouseWheel(SInt32 xDelta, SInt32 yDelta) const = 0;
      |                ^
In file included from /usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/lib/platform/XWindowsScreen.cpp:19:
/usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/./lib/platform/XWindowsScreen.h:79:16: error: 'enable' overrides a member function but is not marked 'override' [-Werror,-Winconsistent-missing-override]
   79 |   virtual void enable();
      |                ^
/usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/./lib/synergy/PlatformScreen.h:88:16: note: overridden virtual function is here
   88 |   virtual void enable() = 0;
      |                ^
In file included from /usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/lib/platform/XWindowsScreen.cpp:19:
/usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/./lib/platform/XWindowsScreen.h:80:16: error: 'disable' overrides a member function but is not marked 'override' [-Werror,-Winconsistent-missing-override]
   80 |   virtual void disable();
      |                ^
/usr/ports/sysutils/synergy/work/synergy-1.15.1-r1/src/./lib/synergy/PlatformScreen.h:89:16: note: overridden virtual function is here
   89 |   virtual void disable() = 0;
      |                ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
```
```


### Additional information

_No response_

**Comment by nbolton** [comment]
Thanks for the report. I plan on getting a FreeBSD runner setup, but since GitHub doesn't support FreeBSD, I'm considering perhaps using QEMU (maybe with https://github.com/docker/setup-qemu-action or something like that).

**Comment by yurivict** [comment]
I've never set up FreeBSD CI on GitHub, but the votca project runs FreeBSD CI successfully:
https://github.com/votca/votca/actions/runs/10626679257

**Comment by nbolton** [comment]
Oh great, this looks really interesting! https://github.com/vmactions/freebsd-vm

**Comment by nbolton** [comment]
@yurivict Hey Yuri :wave: 

So, I noticed that even though we're setting:
```
set(CMAKE_INCLUDE_PATH "${CMAKE_INCLUDE_PATH}:/usr/local/include")
```

That isn't working for me anymore and I was getting the error:
```
  CMake Error at cmake/Libraries.cmake:341 (message):
    Missing header: X11/Xlib.hX11/XKBlib.h
  Call Stack (most recent call first):
    cmake/Libraries.cmake:109 (configure_xorg_libs)
    cmake/Libraries.cmake:8 (configure_unix_libs)
    CMakeLists.txt:28 (configure_libs)
```

I'm wondering if you saw that error or if it was just me.

To solve it, I had to set `CMAKE_REQUIRED_INCLUDES` instead:
```
set(CMAKE_REQUIRED_INCLUDES ${CMAKE_REQUIRED_INCLUDES} "/usr/local/include")
```

Ok... well that fixed the failing `check_include_files` calls, but now I'm seeing an error linking to Xtst even though the file is definitely there.

```
  CMake Error at cmake/Libraries.cmake:377 (message):
    Missing library: Xtst
  Call Stack (most recent call first):
    cmake/Libraries.cmake:109 (configure_xorg_libs)
    cmake/Libraries.cmake:8 (configure_unix_libs)
    CMakeLists.txt:28 (configure_libs)
  -- Configuring incomplete, errors occurred!
```

I'll keep plugging away at it and I'll get there eventually, but any tips from an experienced FreeBSD user would be greatly appreciated! :smile: 

I was sure that setting `CMAKE_REQUIRED_LIBRARIES` would do the trick but haven't had much luck so far.

**Comment by nbolton** [comment]
Well, after a few hours of mucking about with CMake vars, turns out this is the magic line:
```
set(CMAKE_REQUIRED_FLAGS "-L/usr/local/lib")
```

**Comment by nbolton** [comment]
I wanted to share this beautiful sight.

![image](https://github.com/user-attachments/assets/b8771743-e936-49a0-997f-38466c577fa8)

**Comment by nbolton** [comment]
@yurivict Comment for you: https://github.com/symless/synergy/pull/7474#issuecomment-2332717758

**Comment by nbolton** [comment]
@yurivict I'm going to close this as I believe it's fixed but let me know if you still have problems. To keep things simple, this will be released in 1.16, but if you need me to back-port the fix to 1.15 and do a patch release, just let me now.
